lib/hcloud/entry_loader.rb

```diff
@@ -77,17 +77,17 @@ def from_response(response, autoload_action: nil)
         action = response.parsed_json[:action] if autoload_action
         client = response.context.client
         if attributes.is_a?(Array)
-          results = attributes.map { |item| new(item).tap { |entity| entity.response = response } }
+          results = attributes.map { |item| new(client, item).tap { |entity| entity.response = response } }
           results.tap { |ary| ary.extend(Collection) }.response = response
           return results
         end
 
         return Action.new(client, action) if attributes.nil? && action
-        return new(attributes).tap { |entity| entity.response = response } if action.nil?
+        return new(client, attributes).tap { |entity| entity.response = response } if action.nil?
 
         [
           Action.new(client, action),
-          new(attributes).tap { |entity| entity.response = response }
+          new(client, attributes).tap { |entity| entity.response = response }
```

Comment:
Why is the client missing in the `new`? I mean the `initialize` with intercharchible arguments is quite complex so I guess there is a reason? @tonobo 

This led to Future objects having nil as client.

Reply:
Well there is no need for passing the client here directly. The client will be used from response context in most cases, as far as i see.

Reply:
Ah ok got it. You're right. Setting the response right after `new()` it a bit to late. So the client is not picked from the reference to response as it's not set up during this time. This means it's pointing to nil `@client` and keeps passing this over to attached resources. Good catch :+1: